src/pid/limits.rs

```diff
@@ -0,0 +1,335 @@
+//! Process limits informations from `/proc/[pid]/limits`.
+
+use std::fs::File;
+use std::io::Result;
+use std::str::{self};
+use time::Duration;
+
+use libc::pid_t;
+use nom::{
+    IResult,
+    line_ending,
+    is_space
+};
+
+use parsers::{
+    map_result,
+    parse_isize,
+    parse_word,
+    read_to_end
+};
+
+/// Parses a limit value
+/// turns it into an isize for later processing
+named!(parse_limit_value<&[u8], isize>,
+    alt!(map!(tag_s!("unlimited"), |_| LIMITS_INFINITY) | parse_isize)
+);
+
+/// Parses a limit line
+named!(parse_limit_line<&[u8],(isize,isize,Option<LimitUnit>)>,
+    do_parse!(
+        // We don't care about the limit name
+        take_until!("  ") >>
+        // Take while we hit a space
+        take_while!(is_space) >>
+        // Parses soft limit
+        soft_limit: parse_limit_value >>
+        // Take while we hit a space
+        take_while!(is_space) >>
+        // Parses hard limit
+        hard_limit: parse_limit_value >>
+        // Take while we hit a space
+        take_while!(is_space) >>
+        // Parses the unit of the limit. The value may not be specified
+        unit: map!(opt!(parse_word), unit_types) >>
+        // Take while we hit a space
+        take_while!(is_space) >>
+        // End of the limit line
+        line_ending >>
+
+        ((soft_limit, hard_limit, unit))
+    )
+);
+
+/// A constant to represent the "unlimited" value
+pub const LIMITS_INFINITY: isize = -1;
+
+/// A limit unit Enum. Has a field for every unit we may encounter (according to Linux version)
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum LimitUnit{
```

Comment:
I'm a little bit skeptical that this enum is adding any value.  The unit should be unambiguous from the field name.  One thing we have done in other proc API parsers is to convert to a standard base when a unit has magnitude, so for instance we always present byte amounts as a usize, even if it were provided as something like "1kb".